audio/data.py

```diff
@@ -214,12 +232,19 @@ def _pre_process(self):
         
         if len(x.items) > 0:
             if not cfg.resample_to: _set_sr(x.items[0], x.config, x.path)
-            if cfg.remove_silence or cfg.segment_size or cfg.resample_to:
+            _set_nchannels(x.items[0], x.config)
```

Comment:
We only need to call this function once, so wrap it in a statement that checks if config._nchannels is already set...
`if cfg._nchannels is None: _set_nchannels(x.items[0], x.config)`